api/filter_dimensions.go

```diff
@@ -6,20 +6,39 @@ import (
 	"fmt"
 	"github.com/ONSdigital/dp-filter-api/filters"
 	"github.com/ONSdigital/dp-filter-api/models"
+	"github.com/ONSdigital/go-ns/common"
 	"github.com/ONSdigital/go-ns/log"
 	"github.com/gorilla/mux"
 	"net/http"
 )
 
-func (api *FilterAPI) getFilterBlueprintDimensions(w http.ResponseWriter, r *http.Request) {
+const (
+	// audit actions
+	getDimensionsAction   = "getFilterBlueprintDimensions"
+	getDimensionAction    = "getFilterBlueprintDimension"
+	removeDimensionAction = "removeFilterBlueprintDimension"
+	addDimensionAction    = "addFilterBlueprintDimension"
+)
+
+func (api *FilterAPI) getFilterBlueprintDimensionsHandler(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
-	filterID := vars["filter_blueprint_id"]
-	logData := log.Data{"filter_blueprint_id": filterID}
+	filterBlueprintID := vars["filter_blueprint_id"]
+	logData := log.Data{"filter_blueprint_id": filterBlueprintID}
 	log.Info("getting filter blueprint dimensions", logData)
 
-	filter, err := api.getFilterBlueprint(r.Context(), filterID)
+	auditParams := common.Params{"filter_blueprint_id": filterBlueprintID}
+	if auditErr := api.auditor.Record(r.Context(), getDimensionsAction, actionAttempted, auditParams); auditErr != nil {
+		handleAuditingFailure(r.Context(), getDimensionsAction, actionAttempted, w, auditErr, logData)
+		return
+	}
+
+	filter, err := api.getFilterBlueprint(r.Context(), filterBlueprintID)
 	if err != nil {
 		log.ErrorC("unable to get dimensions for filter blueprint", err, logData)
+		if auditErr := api.auditor.Record(r.Context(), getDimensionsAction, actionUnsuccessful, auditParams); auditErr != nil {
+			handleAuditingFailure(r.Context(), getDimensionsAction, actionUnsuccessful, w, auditErr, logData)
```

Comment:
actually, the log line might be better _after_ the `if Record ...` conditional - i.e. log only on successful audit, because `handleAuditingFailure` does logging, too

Reply:
The gist I got from previous conversations were that auditing should just identify if the user was able to complete an action. The cause of a failed attempt would be determined by the logs instead of duplicating the failure message in the audit logs.

Regarding the last point about moving the `log.Error*` line: The `handleAuditingFailure ` log is specifically to log the audit failure. The `log.Error*` line is logging the failed action - not relating to auditing.

Reply:
understood